sdk/scheduler/src/test/java/com/mesosphere/sdk/http/EndpointUtilsTest.java

```diff
@@ -33,49 +35,73 @@ public void testToEndpoint() {
-        assertEquals("vip.pathtosvc.with.dots.l4lb.thisdcos.directory:5", EndpointUtils.toVipEndpoint("path/to/svc.with.dots", new VipInfo("vip", 5)));
+        assertEquals("vip.svc.vip.tld:5",
+                EndpointUtils.toVipEndpoint("svc", mockSchedulerConfig, new VipInfo("vip", 5)));
+        assertEquals("vip.pathtosvc.vip.tld:5",
+                EndpointUtils.toVipEndpoint("/path/to/svc", mockSchedulerConfig, new VipInfo("vip", 5)));
+        assertEquals("vip.pathtosvc.vip.tld:5",
+                EndpointUtils.toVipEndpoint("path/to/svc", mockSchedulerConfig, new VipInfo("vip", 5)));
+        assertEquals("vip.pathtosvc.with.dots.vip.tld:5",
+                EndpointUtils.toVipEndpoint("path/to/svc.with.dots", mockSchedulerConfig, new VipInfo("vip", 5)));
 
-        assertEquals("vip.svc.l4lb.thisdcos.directory:5", EndpointUtils.toVipEndpoint("svc", new VipInfo("/vip", 5)));
-        assertEquals("vip.pathtosvc.l4lb.thisdcos.directory:5", EndpointUtils.toVipEndpoint("/path/to/svc", new VipInfo("/vip", 5)));
-        assertEquals("vip.pathtosvc.l4lb.thisdcos.directory:5", EndpointUtils.toVipEndpoint("path/to/svc", new VipInfo("/vip", 5)));
-        assertEquals("vip.pathtosvc.with.dots.l4lb.thisdcos.directory:5", EndpointUtils.toVipEndpoint("path/to/svc.with.dots", new VipInfo("/vip", 5)));
+        assertEquals("vip.svc.vip.tld:5",
+                EndpointUtils.toVipEndpoint("svc", mockSchedulerConfig, new VipInfo("/vip", 5)));
+        assertEquals("vip.pathtosvc.vip.tld:5",
+                EndpointUtils.toVipEndpoint("/path/to/svc", mockSchedulerConfig, new VipInfo("/vip", 5)));
+        assertEquals("vip.pathtosvc.vip.tld:5",
+                EndpointUtils.toVipEndpoint("path/to/svc", mockSchedulerConfig, new VipInfo("/vip", 5)));
+        assertEquals("vip.pathtosvc.with.dots.vip.tld:5",
+                EndpointUtils.toVipEndpoint("path/to/svc.with.dots", mockSchedulerConfig, new VipInfo("/vip", 5)));
     }
 
     @Test
     public void testToSchedulerAutoIpHostname() {
-        assertEquals("svc.marathon.some.tld", EndpointUtils.toSchedulerAutoIpHostname("svc", mockSchedulerConfig));
-        assertEquals("svc-to-path.marathon.some.tld", EndpointUtils.toSchedulerAutoIpHostname("/path/to/svc", mockSchedulerConfig));
-        assertEquals("svc-to-path.marathon.some.tld", EndpointUtils.toSchedulerAutoIpHostname("path/to/svc", mockSchedulerConfig));
-        assertEquals("svc-with-dots-to-path.marathon.some.tld", EndpointUtils.toSchedulerAutoIpHostname("path/to/svc.with.dots", mockSchedulerConfig));
+        assertEquals("svc.test-marathon.autoip.tld",
+                EndpointUtils.toSchedulerAutoIpHostname("svc", mockSchedulerConfig));
+        assertEquals("svc-to-path.test-marathon.autoip.tld",
+                EndpointUtils.toSchedulerAutoIpHostname("/path/to/svc", mockSchedulerConfig));
+        assertEquals("svc-to-path.test-marathon.autoip.tld",
```

Comment:
Would be nice to have this in a separate test that the leading slash doesn't affect the hostname.